### DISCUSSION TITLE
How to get the temporary yaml file that argo applies when deploying, to debug failed deployments.

### DISCUSSION THREAD

**Discussion by dom38** [argoproj/argo-cd#8203] [Q&A]
I'm currently refactoring about 2 years worth of ArgoCD application templates and when applying some of them I get an error like:

```rpc error: code = Unknown desc = Manifest generation error (cached): `helm template . --name-template chart-name --namespace the namespace --kube-version 1.20 --values /tmp/values-some number.yaml --api-versions {one million api versions} --include-crds` failed exit status 1: Error: failed to parse /tmp/values-some number.yaml: error converting YAML to JSON: yaml: line 13: mapping values are not allowed in this context```

There are a few levels of abstraction between the templates I write and the charts being deployed via ArgoCD, so replicating this with the values is a pain (But doable). 

However, I would like to know the contents of the `/tmp/values-some number.yaml` file, every time this error is posted, because that would be a faster way for me to be able to see what has gone wrong in the steps between my application template and Argo deploying the chart. 

Is there a way to fetch the temp files used to deploy resources via the Argo UI? If not via the UI, where can I find these on the filesystem of the pods that are running ArgoCD? I had a look through a few pods and couldn't find the temp file.

Thanks for looking.

**Comment by meln5674** [comment]
The contents of that temporary values file is nothing more than the 'spec.source.helm.values' field within the app itself, and you can verify this in the code: https://github.com/argoproj/argo-cd/blob/4aa614dafb933a2d5c66ed29cc37bd7421e8c01f/reposerver/repository/repository.go#L622 . If you're creating this dynamically using the app of apps pattern, you can view the app resource in its "parent" app in the UI, or you can pull it out using kubectl, e.g.

```bash
kubectl get app your-app --template '{{ .spec.source.helm.values }}' > your-app-values.yaml
```
